Replicator/c4ReplicatorImpl.hh

```diff
@@ -492,16 +492,26 @@ namespace litecore {
                 // calling pendingDocumentIDs (which might call into the app's validation function.)
                 LOCK(repl->_mutex);
                 replicator = repl->_replicator;
-                
+
                 // CBL-2448: Also make my own checkpointer and database in case a call comes in
                 // after Replicator::terminate() is called.  The fix includes the replicator
                 // pending document ID function now returning a boolean success, isDocumentPending returning
                 // an optional<bool> and if pendingDocumentIDs returns false or isDocumentPending
                 // returns nullopt, the checkpointer is fallen back on
-                C4Collection* collection = repl->_database->getCollection(collectionSpec);
-                if (collection != nullptr) {
-                    checkpointer.emplace(repl->_options, repl->URL(), collection);
+                C4Collection* collection = nullptr;
+                // The collection must be included in the replicator's config options.
+                auto it = repl->_options->collectionSpecToIndex().find(collectionSpec);
+                if (it != repl->_options->collectionSpecToIndex().end()) {
+                    if (it->second < repl->_options->workingCollectionCount()) {
+                        collection = repl->_database->getCollection(collectionSpec);
+                    }
+                }
+
+                if (collection == nullptr) {
+                    error::_throw(error::NotOpen, "collection not in the Replicator's config");
```

Comment:
NotFound?

Reply:
From slack : `The fact is, the collection can be found in the database, but not "opened" via the config.`